### PR TITLE
Config v2 overhaul & misc other fixes found while testing it

### PR DESCRIPTION
**Pull request by Artanicus**
Dependent on https://github.com/florianbecker/php.core.telegram/pull/19
- Rework all config usage to config_v2 as per above core pull request.
- Fixes ability to share to level specific chats from \list
- Fixes some weirdness with overviews, introduces new weirdness that is less bad
- Splits sample config into a true sample and default values.
- Existing configs will continue to work but if defaults change, you are likely overriding them in your custom conf!

TL;DR; to update your config to the latest and greatest if you want to:
- remove any lines you don't want to customize and want the default instead
- Change any "true" to true and "false" to false to be more future compatible. There is still a compatibility conversion for now though.

### CODE REVIEW

**Comment by Artanicus** [comment]
Friendly ping on this PR, have you had a chance to take a look at it? The merge is still fine now but will start diverging if parallel changes are made with the old config structure.

**Comment by Artanicus** [comment]
This now contains the fixes that came up in the code review by @Chefkeks
